api/src/main/java/org/openmrs/module/appointments/conflicts/impl/PatientDoubleBookingConflict.java

```diff
@@ -0,0 +1,59 @@
+package org.openmrs.module.appointments.conflicts.impl;
+
+import org.openmrs.module.appointments.conflicts.AppointmentConflictType;
+import org.openmrs.module.appointments.dao.AppointmentDao;
+import org.openmrs.module.appointments.model.Appointment;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.openmrs.module.appointments.constants.AppointmentConflictTypeEnum.PATIENT_DOUBLE_BOOKING;
+
+public class PatientDoubleBookingConflict implements AppointmentConflictType {
+
+    private AppointmentDao appointmentDao;
+
+    public void setAppointmentDao(AppointmentDao appointmentDao) {
+        this.appointmentDao = appointmentDao;
+    }
+
+    @Override
+    public String getType() {
+        return PATIENT_DOUBLE_BOOKING.name();
+    }
+
+    @Override
+    public List<Appointment> getAppointmentConflicts(List<Appointment> appointments) {
+        List<Appointment> conflictingAppointments = new ArrayList<>();
+        List<Appointment> patientAppointments = getPatientAppointments(appointments.get(0).getPatient().getPatientId());
+        for (Appointment appointment : appointments) {
+            conflictingAppointments.addAll(getConflictingAppointments(appointment, patientAppointments));
+        }
+        return conflictingAppointments;
```

Comment:
Does `conflictingAppointments ` can hold duplicate appointments by any chance?

Reply:
The only case it can hold duplicate is when two appointments are conflicting with same appointment, which is almost zero case